test/api/mm/testers/test_fileinitc.c

```diff
@@ -40,37 +40,40 @@ int main()
   double *gradients, *gross_charges;
   int natom;
 
-  /* Initialise DFTB+ input tree from input in external file */
-  dftbp_init(&calculator, NULL);
-  dftbp_get_input_from_file(&calculator, "dftb_in.hsd", &input);
-  dftbp_process_input(&calculator, &input);
-
-  /* Check whether the calculator was initialized with the correct nr. of atoms */
-  natom = dftbp_get_nr_atoms(&calculator);
-  printf("Expected nr. of atoms: %d\n", NR_OF_ATOMS);
-  printf("Obtained nr. of atoms: %d\n", natom);
-
-  /* Override coordinates in the tree*/
-  dftbp_set_coords_and_lattice_vecs(&calculator, coords, latvecs);
-
-  /* evaluate energy */
-  dftbp_get_energy(&calculator, &mermin_energy);
-  printf("Expected Mermin-energy: %15.10f\n", -2.5933460731);
-
-  /* and gradients */
-  gradients = (double *) calloc(3 * NR_OF_ATOMS, sizeof(double));
-  dftbp_get_gradients(&calculator, gradients);
-  printf("Expected gradient of atom 1: %15.10f %15.10f %15.10f\n", -0.010321385989, -0.010321385989,
-         -0.010321385989);
-
-  /* and gross charges */
-  gross_charges = (double *) calloc(NR_OF_ATOMS, sizeof(double));
-  dftbp_get_gross_charges(&calculator, gross_charges);
-  printf("Obtained gross charges: %15.10f %15.10f\n", gross_charges[0], gross_charges[1]);
-
-  /*  clean up */
-  dftbp_final(&calculator);
-
+  /* Dummy loop to test subsequent initialisations of the DFTB+ object */
+  for (int ii = 0; ii < 2; ++ii) {
+  
+    /* Initialise DFTB+ input tree from input in external file */
+    dftbp_init(&calculator, NULL);
+    dftbp_get_input_from_file(&calculator, "dftb_in.hsd", &input);
+    dftbp_process_input(&calculator, &input);
+
+    /* Check whether the calculator was initialized with the correct nr. of atoms */
+    natom = dftbp_get_nr_atoms(&calculator);
+    printf("Expected nr. of atoms: %d\n", NR_OF_ATOMS);
+    printf("Obtained nr. of atoms: %d\n", natom);
+
+    /* Override coordinates in the tree*/
+    dftbp_set_coords_and_lattice_vecs(&calculator, coords, latvecs);
+
+    /* evaluate energy */
+    dftbp_get_energy(&calculator, &mermin_energy);
+    printf("Expected Mermin-energy: %15.10f\n", -2.5933460731);
+
+    /* and gradients */
+    gradients = (double *) calloc(3 * NR_OF_ATOMS, sizeof(double));
+    dftbp_get_gradients(&calculator, gradients);
+    printf("Expected gradient of atom 1: %15.10f %15.10f %15.10f\n", -0.010321385989,
```

Comment:
Probably better to test with different cases in the iteration of the loop, in case of any internal state surviving the dftb_final() call.

Reply:
Good point. I have did that now in 978f8 and the code still seems to work without changes.